plugins/techdocs/src/reader/components/TechDocsReaderPageContent/dom.tsx

```diff
@@ -64,50 +70,53 @@ export const useTechDocsReaderDom = (
 ): Element | null => {
   const navigate = useNavigate();
   const theme = useTheme<BackstageTheme>();
+
   const techdocsStorageApi = useApi(techdocsStorageApiRef);
   const scmIntegrationsApi = useApi(scmIntegrationsApiRef);
   const techdocsSanitizer = useApi(configApiRef);
+
   const { namespace, kind, name } = entityRef;
   const { state, path, content: rawPage } = useTechDocsReader();
+
   const isDarkTheme = theme.palette.type === 'dark';
 
-  const [sidebars, setSidebars] = useState<HTMLElement[]>();
   const [dom, setDom] = useState<HTMLElement | null>(null);
+  const isDomLoading = useShadowDomStylesLoading(dom);
 
   // sidebar pinned status to be used in computing CSS style injections
   const { isPinned } = useContext(SidebarPinStateContext);
 
   const updateSidebarPosition = useCallback(() => {
-    if (!dom || !sidebars) return;
-    // set sidebar height so they don't initially render in wrong position
-    const mdTabs = dom.querySelector('.md-container > .md-tabs');
-    const sidebarsCollapsed = window.matchMedia(
-      'screen and (max-width: 76.1875em)',
-    ).matches;
-    const newTop = Math.max(dom.getBoundingClientRect().top, 0);
+    if (!dom) return;
+
+    const sidebars = dom.querySelectorAll<HTMLElement>('.md-sidebar');
+
+    const isMobile = window.matchMedia(MOBILE_MEDIA_QUERY).matches;
+
     sidebars.forEach(sidebar => {
-      if (sidebarsCollapsed) {
+      // set sidebar height so they don't initially render in wrong position
+      if (isMobile) {
         sidebar.style.top = '0px';
-      } else if (mdTabs) {
-        sidebar.style.top = `${
-          newTop + mdTabs.getBoundingClientRect().height
-        }px`;
       } else {
-        sidebar.style.top = `${newTop}px`;
+        const domTop = dom.getBoundingClientRect().top ?? 0;
+        const tabs = dom.querySelector('.md-container > .md-tabs');
+        const tabsHeight = tabs?.getBoundingClientRect().height ?? 0;
+        sidebar.style.top = `${Math.max(domTop, 0) + tabsHeight}px`;
       }
+
+      // Show the sidebar only after updating its position
+      sidebar.style.setProperty('opacity', '1');
```

Comment:
Changing the position of the sidebars was previously perceived by the user, now we only display the sidebars after their positions have been updated.

See how it was before, you can see the position of the sidebars being changed from top to bottom:
![sidebar-bug](https://user-images.githubusercontent.com/6290749/168607742-520ef38a-d250-4ef1-98d3-07ca804ae0d6.gif)